apis/hive/v1/agentclusterinstall_types.go

```diff
@@ -0,0 +1,170 @@
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AgentClusterInstall represents a request to provision an agent based cluster.
+//
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+type AgentClusterInstall struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AgentClusterInstallSpec   `json:"spec"`
+	Status AgentClusterInstallStatus `json:"status,omitempty"`
+}
+
+// HyperthreadingMode is the mode of hyperthreading for a machine.
+// +kubebuilder:validation:Enum="";Enabled;Disabled
+type HyperthreadingMode string
+
+const (
+	// HyperthreadingEnabled indicates that hyperthreading is enabled.
+	HyperthreadingEnabled HyperthreadingMode = "Enabled"
+	// HyperthreadingDisabled indicates that hyperthreading is disabled.
+	HyperthreadingDisabled HyperthreadingMode = "Disabled"
+)
+
+// AgentClusterInstallSpec defines the desired state of the AgentClusterInstall.
+type AgentClusterInstallSpec struct {
+
+	// ImageSetRef is a reference to a ClusterImageSet. The release image specified in the ClusterImageSet will be used
+	// to install the cluster.
+	ImageSetRef ClusterImageSetReference `json:"imageSetRef"`
+
+	// ClusterDeploymentRef is a reference to the ClusterDeployment associated with this AgentClusterInstall.
+	ClusterDeploymentRef corev1.LocalObjectReference `json:"clusterDeploymentRef"`
+
+	// ClusterMetadata contains metadata information about the installed cluster. It should be populated once the cluster install is completed. (it can be populated sooner if desired, but Hive will not copy back to ClusterDeployment until the Installed condition goes True.
+	ClusterMetadata *ClusterMetadata `json:"clusterMetadata,omitempty"`
+
+	// ManifestsConfigMapRef is a reference to user-provided manifests to
+	// add to or replace manifests that are generated by the installer.
+	ManifestsConfigMapRef *corev1.LocalObjectReference `json:"manifestsConfigMapRef,omitempty"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking Networking `json:"networking"`
+
+	// SSHPublicKey will be added to all cluster hosts for use in debugging.
+	// +optional
+	SSHPublicKey string `json:"sshPublicKey,omitempty"`
+
+	// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+	ProvisionRequirements ProvisionRequirements `json:"provisionRequirements"`
+
+	// ControlPlane is the configuration for the machines that comprise the
+	// control plane.
+	// +optional
+	ControlPlane *AgentMachinePool `json:"controlPlane,omitempty"`
+
+	// Compute is the configuration for the machines that comprise the
+	// compute nodes.
+	// +optional
+	Compute []AgentMachinePool `json:"compute,omitempty"`
+}
+
+// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+type ProvisionRequirements struct {
```

Comment:
```suggestion
type AgentProvisionRequirements struct {
```

or move this to hive/v1/agent package

Reply:
done.